src/main/java/uk/gov/companieshouse/lookup/config/MessageConfig.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.companieshouse.lookup.config;
+
+import java.util.Locale;
+
+import org.springframework.context.MessageSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.web.servlet.LocaleResolver;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.i18n.LocaleChangeInterceptor;
+import uk.gov.companieshouse.lookup.internationalisation.ChSessionLocaleResolver;
+
+@Configuration
+public class MessageConfig implements WebMvcConfigurer {
+    @Bean("messageSource")
+    public MessageSource messageSource() {
+        var messageSource = new ResourceBundleMessageSource();
+
+        // This will match the resources folder files that begin with 'messages'
+        // e.g. messages_wl.properties, messages_fr.properties
+        messageSource.setBasenames("locales/messages", "locales/common-messages");
+        messageSource.setDefaultEncoding("UTF-8");
+        return messageSource;
+    }
+
+    // The LocaleResolver interface has implementations that determine
+    // the current locale based on the session, cookies,
+    // the Accept-Language header, or a fixed value.
+    @Bean
+    public LocaleResolver localeResolver(ChSessionLocaleResolver chSessionLocaleResolver) {
+        chSessionLocaleResolver.setDefaultLocale(Locale.ENGLISH);
```

Comment:
Where is ChSessionLocaleResolver being injected from?

Reply:
This ChSessionLocaleResolver CH's recent addition to master to override the default Spring LocaleResolver with their own implementation which handles interactions with the CH session, specifically it:

1. Checks if the current session is a [Welsh language enabled journey](https://github.com/companieshouse/company-lookup.web.ch.gov.uk/blob/27510a8db9c763bb171e684368bce5b7cf66380e/src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java#L29), acting as a feature flag
2. [Handles](https://github.com/companieshouse/company-lookup.web.ch.gov.uk/blob/27510a8db9c763bb171e684368bce5b7cf66380e/src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java#L42) [interactions](https://github.com/companieshouse/company-lookup.web.ch.gov.uk/blob/27510a8db9c763bb171e684368bce5b7cf66380e/src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java#L60) with the session to store the language state so it persists along a Welsh enabled journey

It's defined as a [@Component](https://github.com/companieshouse/company-lookup.web.ch.gov.uk/blob/27510a8db9c763bb171e684368bce5b7cf66380e/src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java#L14) meaning it's available and managed by spring

Reply:
Thanks @dhawes-ch. Why can't this flag be set in the place where ChSessionLocaleResolver is created?